katapult.go

```diff
@@ -92,13 +103,70 @@ func New(opts ...Opt) (*Client, error) {
 	return c, nil
 }
 
-type Client struct {
-	HTTPClient *http.Client
-	Codec      codec.Codec
+func (c *Client) DoIt(
+	ctx context.Context,
+	request *Request,
+	v interface{},
+) (*Response, error) {
+	contentType, bodyReader, err := request.bodyContent()
+	if err != nil {
+		return nil, err
+	}
 
-	APIKey    string
-	UserAgent string
-	BaseURL   *url.URL
+	u := c.BaseURL.ResolveReference(request.URL)
+	req, err := http.NewRequestWithContext(
+		ctx, request.Method, u.String(), bodyReader,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(request.Header) > 0 {
+		for k := range request.Header {
+			for _, v := range request.Header.Values(k) {
+				req.Header.Add(k, v)
+			}
+		}
+	}
+
+	if !request.NoAuth {
+		if c.APIKey == "" {
+			return nil, fmt.Errorf(
+				"%w: no API key available for: %s %s",
+				ErrRequest, request.Method, request.URL.Path,
+			)
+		}
+		req.Header.Set(
+			"Authorization",
+			fmt.Sprintf("Bearer %s", c.APIKey),
+		)
+	}
+	req.Header.Set("User-Agent", c.UserAgent)
+	req.Header.Set("Accept", "application/json")
+	if contentType != "" {
+		req.Header.Set("Content-Type", contentType)
+	}
+
+	r, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return NewResponse(nil), err
+	}
+	defer r.Body.Close()
+
+	resp := NewResponse(r)
+	if resp.StatusCode/100 != 2 {
```

Comment:
Are we sure 100/ 300 status codes aren't used meaningfully in the API as it stands?

Reply:
As far as I know success responses are always `2xx` of some kind.